tests/unit/models/ObjectMD.js

```diff
@@ -257,6 +257,23 @@ describe('ObjectMD class setters/getters', () => {
         md.clearMetadataValues();
         assert.strictEqual(md.getUserMetadata(), undefined);
     });
+
+    it('ObjectMD::microVersionId', () => {
+        assert.strictEqual(md.getMicroVersionId(), null);
+
+        const generatedIds = {};
+        for (let i = 0; i < 100; ++i) {
+            md.updateMicroVersionId();
+            generatedIds[md.getMicroVersionId()] = 1;
+        }
+        // all generated IDs should be different
+        assert.strictEqual(Object.keys(generatedIds).length, 100);
```

Comment:
Although unlikely this does have a chance to fail due to getting the same random 8 bytes out of 100, I am sure the odds are incredibly tiny but I would also argue that this doesn't really validate our code much either. Simply mocking the random call will have the same affect here.

Reply:
The idea of this test is to ensure that we get a different ID every time it's called, it's more of a smoke test than anything else.

I'm not convinced mocking the random call would add much value, since there is also value in running the actual random code that is being used in production, and it's possible to call it within a unit test.

Reply:
Basically by mocking an existing method we piggyback on all the testing and validation of that method ( or anything really but here its a method ). `crypto.randomBytes(8)` is going to be a very well tested method, simply by mocking and asserting we called that method is plenty of value to me.

Creating a smoke test, or length check or any repeated test here is going to be much less exhaustive when compared to what the crypto package already has to go through... probably. So I would argue being lazy here has the most value.

tl;dr having a test that says we call the crypto package is pretty convincing to me its well tested.